packages/cli/src/push/index.ts

```diff
@@ -91,25 +102,8 @@ export class Push extends TypedEmitter<PushEvents> {
 	}
 
 	send(pushMsg: PushMessage, pushRef: string) {
-		const { isWorker, isMultiMain } = this.instanceSettings;
-
-		/**
-		 * In scaling mode, in single- or multi-main setup, in a manual execution,
-		 * a worker relays execution lifecycle events to all mains. Only the main
-		 * who holds the session for the execution will push to the frontend who
-		 * commissioned the execution.
-		 *
-		 * In scaling mode, in multi-main setup, in a manual webhook execution, if
-		 * the main who handles a webhook is not the main who created the webhook,
-		 * the handler main relays execution lifecycle events to all mains. Only
-		 * the main who holds the session for the execution will push events to
-		 * the frontend who commissioned the execution.
-		 */
-		if (isWorker || (isMultiMain && !this.hasPushRef(pushRef))) {
-			void this.publisher.publishCommand({
-				command: 'relay-execution-lifecycle-event',
-				payload: { ...pushMsg, pushRef },
-			});
+		if (this.shouldRelayViaPubSub(pushRef)) {
+			this.relayViaPubSub(pushMsg, pushRef);
```

Comment:
IMHO this logic shouldn't be in the `Push` class. Now it ties the logic to implementation details of the scaling mode, where as Push should only be concerned about communicating via the websocket (or SSE). Now since this has already been here before we don't have to move it elsewhere in this PR. I just wanted to mention this for future

Reply:
Agree, we need to work on the lifecycle hooks first to unlock this refactor.